example/lib/main.dart

```diff
@@ -47,12 +43,64 @@ class _MyAppState extends State<MyApp> {
   @override
   Widget build(BuildContext context) {
     return MaterialApp(
+      debugShowCheckedModeBanner: false,
       home: Scaffold(
         appBar: AppBar(
-          title: const Text('Plugin example app'),
+          title: const Text(
+            ExampleStrings.appBarTitle,
+          ),
         ),
         body: Center(
-          child: Text('Running on: $_platformVersion\n'),
+          child: Column(
+            mainAxisAlignment: MainAxisAlignment.spaceEvenly,
+            children: [
+              Column(
+                children: [
+                  PluginButton(
+                    text: ExampleStrings.checkPermission,
+                    onPressed: () {
+                      AndroidLocatorPlugin.checkPermission;
+                    },
+                  ),
+                  StreamBuilder(
+                    stream: AndroidLocatorPlugin.accessStream,
+                    builder: (
+                      BuildContext context,
+                      AsyncSnapshot<dynamic> snapshot,
+                    ) {
+                      return snapshot.hasData
+                          ? Text(
+                              snapshot.data.toString(),
+                            )
+                          : Text(
+                              ExampleStrings.permissionInfo,
+                            );
+                    },
+                  ),
+                ],
+              ),
+              Column(
+                children: [
+                  PluginButton(
+                    text: ExampleStrings.requestPermission,
+                    onPressed: () {
+                      AndroidLocatorPlugin.requestPermission;
+                    },
+                  ),
+                ],
+              ),
+              Column(
+                children: [
+                  PluginButton(
+                    text: ExampleStrings.initializePlugin,
+                    onPressed: () {
+                      AndroidLocatorPlugin.initializePlugin;
+                    },
+                  ),
```

Comment:
Note that you are doing the same two times, you repeat the same structure, this can be modify creating a new widget that has two parameters for creation, one is relative to text that you show and the second one is the action that you call on onPressed